metrics.go

```diff
@@ -129,6 +133,14 @@ func (m *metrics) start() error {
 		return err
 	}
 
+	if m.unindexedDealCount, err = meter.AsyncInt64().Gauge(
+		"ipni/heyfil/state_market_unindexed_deal_count",
```

Comment:
maybe use `prometheus.BuildFQName` to build these?

Reply:
Thanks, I will reflect on this in a separate PR. The otel library used by heyfil has since been upgraded with breaking changes, which also needs to be reflected.